### ISSUE TITLE
HABs layer to MTRI remote sensing

### ISSUE THREAD

**Issue by kkoch** [glos/myglos#169]
From Colin >

Hi DMAC colleagues - a remaining item on our statement of work for this year's DMAC activities is sharing the Harmful Algal Bloom product that MTRI creates (under NOAA GLERL funding). We've been working on this derived product for a while, and feel its ready to share now through the GLOS Data Portal. With the prominence of this year's HAB in western Lake Erie, this also seems like a good time to get it shared.

Mike Billmire gave me the technical details to share it via WMS, saying:
"The layer code for the new remote sensing layer is habs; like the kd490 and pzd products, to display this data just use this code in place of chl, cdom, doc, etc. in the WMS calls.

The one complication is that this layer requires a custom legend (i.e. it does not use the default legend returned by the GetLegendGraphic request like the other layers). Instead, we've attached a static legend graphic that should be used with this layer."

The static legend is attached. Also included below is a screen grab showing how the layer displays on our own remote sensing portal at www.greatlakesremotesensing.org for the most recent version of 9/26/17.

![image](https://user-images.githubusercontent.com/6392982/31667797-6fff9f4a-b31e-11e7-90e2-f7ad2c018fe7.png)

![mtri_habs_legend](https://user-images.githubusercontent.com/6392982/31667969-e901dd5e-b31e-11e7-9f99-35e5a2ea2caf.png)

**Comment by kkoch** [comment]
Here's the metadata to go with the MTRI HABs layer info:

"Mapping of remote sensing-derived harmful algal blooms (HABs) products for Lake Erie (Sayers et al. 2016). Imagery consists of two overlapping products. The HAB Related Water Quality Concern product in yellow shows areas where the Color Producing Agent Algorithm (described in Shuchman et al. 2013 and Shuchman et al. 2006) estimates chlorophyll concentrations to be greater than 20 micrograms per Liter. This product is displayed on a color scale of lower concern (dark yellow) to higher concern (yellow) based on the estimated concentration. The HAB Related Public Health Concern product in red shows areas where the normalized difference vegetation index (NDVI, using MODIS bands centered at 667 nm and 859 nm) is greater than zero, indicating that a surface scum is present. This product is displayed on a scale of lower concern (dark red) to higher concern (light red/pink) based on the calculated NDVI value. These layers are processed for Lake Erie approximately once a week during the HABs season (when water temperature is greater than 18 degrees Celsius, approximately early-June to mid-October each year), shared via ftp and WMS with GLOS by the Michigan Tech Research Institute (MTRI) http://www.mtri.org "

If there's room to include the full Sayers et al. 2016 reference as well (it could go at the end), it is:
"Sayers, M., Fahnenstiel, G.L., Shuchman, R.A. and Whitley, M., 2016. Cyanobacteria blooms in three eutrophic basins of the Great Lakes: a comparative analysis using satellite remote sensing. International Journal of Remote Sensing, 37(17), pp.4148-4171."

**Comment by Bobfrat** [comment]
In looking at the GetCapabilities XML, I see the habs layer is only available for Erie, Huron, and Michigan. Is that accurate?

**Comment by kknee** [comment]
@Bobfrat  Yes, looks to be accurate given how the layer is presented on www.greatlakesremotesensing.org

**Comment by Bobfrat** [comment]
You can view these layers on the dev portal now. It looks like the abstracts for individual layers is overwritten by the parent abstract. I'll have to look into this or we update the abstract we had for the other MTRI layers to include some info for HABS

**Comment by ColinNBrooks** [comment]
The display looks like its working well. It would be good to keep the HABs abstract separate if possible - it's quite a different layer. One thing I noticed is that while the MTRI HABs layer shows up in the dev Map view, I couldn't get it to show up in the devCatalog view of the data. If it appears there, I would consider this complete.

**Comment by Bobfrat** [comment]
@ColinNBrooks The layer falls under the MTRI Remote Sensing Lake Erie/Huron/Michigan entries in the catalog. Do you want this layer to be its own entry? That would be one way to get the separate abstract.

**Comment by ColinNBrooks** [comment]
I do think we should do a separate entry - it's a different enough layer that should have its own description (for each lake). I engaged with the layer creators and they wrote the new description for just the MTRI HABs layers.

**Comment by Bobfrat** [comment]
Ok any suggestions for the layer name? The other layers fall under a parent entry named ```MTRI Remote Sensing Lake <name>```. Maybe same format with the more descriptive HABS somewhere in there?

![screen shot 2017-11-06 at 7 37 46 am](https://user-images.githubusercontent.com/5702672/32441428-9c05dd8c-c2c5-11e7-8fc0-f181e71c14bc.png)

**Comment by ColinNBrooks** [comment]
That works well, except I wouldn't capitalize the S (ie, list them as HABs instead of HABS) or just put HAB since it stands for Harmful Algal Blooms. Thanks!

**Comment by kkoch** [comment]
Yep what @ColinNBrooks said.  I think I like HABs.

**Comment by Bobfrat** [comment]
Ok great, I deployed to dev. Please have a look

**Comment by kkoch** [comment]
Maybe it is just me but I can see where the way this displays and how users will find it displayed might be problemmatic.  But I'd like @tslawecki, @kknee  and of course @ColinNBrooks to chime in here.  
1) First of all, because of the way the search engine works, there is no way I can just display the mtri habs views (e.i., I can't do an "and" which is something in my bucket list of wants that I lost out on a long time ago - in other words mtri habs gives me mtri "or" habs because the words aren't right next to each other, I'd have to know to just search mtri or mtri remote sensing).  
2) Then if I find what I want to view, it is so obscured by "other" things that are way more obvious. For example, Lake Michigan.  I don't even notice that there are little tiny bits up there near Green Bay unless I zoom way in because the map doesn't zoom me right there.  In addition, the bounding boxes cover the entire GL region so the user wouldn't even know to zoom in.  It just looks like a map with some big green boxes and buoys.  Huron isn't much better, Erie is somewhat better just because there is more to see.

Screenshot is after I searched for mtri remote sensing and then chose Lake Michigan, find on map. It is even uglier if all I searched for was habs.

![image](https://user-images.githubusercontent.com/6392982/32516845-b5f0dfcc-c3d2-11e7-98d5-a65e05a65bbb.png)

**Comment by Bobfrat** [comment]
I will fix the bounding box issue now. Was inadvertently grabbing the parent layer bounding box

**Comment by kkoch** [comment]
On dev. Will be fixed with next push to production.

**Comment by kkoch** [comment]
@kknee any reason we can't schedule a dev to production push?

**Comment by kknee** [comment]
@Bobfrat I can't think of a reason not to push, let's do it

**Comment by kkoch** [comment]
I think we might have lost some things when we rolled back OWS?  MTRI HABs shows on a search Dev but not production Portal.  HOWEVER, it also doesn't seem to do anything so maybe an effect of turning off MTRI that we didn't consider.  We probably want to add this to our revisitation of MTRI things on May 1.

**Comment by kknee** [comment]
@kkoch we'll look into what's happening on production.  It does do something on dev, check out Lake Erie.
